### DISCUSSION TITLE
Smoke test post deploy script

### DISCUSSION THREAD

**Discussion by janhesters** [vercel/vercel#5780] [Help]
Is there a way to run post deploy smoke tests?

I want to run tests on my environment after it has deployed, and rollback to the latest version, if the deployment failed.

I was able to do this by running a GH action after merge that uses the deploy hooks, but the problem here is, that race conditions happen, when two branches merge in rapid succession.

Is this feature on the roadmap for the GH integration?

**Comment by followbl** [comment]
@janhesters would you mind sharing how you accomplished this - I basically want to do the exact same thing you're describing.

1. Deploy
2. Migrate
3. Smoke

Rollback if anything fails...

Right now I have a GH Action that runs on successful merge to production branch 
-> deploys via [amondnet/vercel-action](https://github.com/amondnet/vercel-action) 
-> migrations run (if any)
-> smoke-test runs

the only things I don't have fully solved yet are...

1. How can I build a production deployment without auto-alias to prod
2. Writing a script to parse the output of `vercel inspect www.production.com` unless there's an easier way to get the currently assigned production alias.

would be interested to hear how you solved this. I did get a ping on a question back in September from @paulogdm that this was on the roadmap - I wonder if there's been any progress on this - as it would be a super helpful flow I believe to a bunch of teams.
